test-proxy.js

```diff
@@ -0,0 +1,111 @@
+import fs from 'fs';
+import test from 'ava';
+import pinkiePromise from 'pinkie-promise';
+import fn from './proxy';
```

Comment:
That's ok for now. We'll either replace proxy with the current approach or adopt both in the same API once we can target Node.js 6.